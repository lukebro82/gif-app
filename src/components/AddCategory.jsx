import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setinputValue] = useState("");

  const onInputChange = ({ target }) => {
    setinputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return; // si no escribo (o una letra sola) nada no hace nada
    setinputValue(""); //borra lo que escribi antes
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={(event) => onSubmit(event)}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
