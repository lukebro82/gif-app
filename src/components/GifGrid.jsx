import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category, onDeleteCategory }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      {" "}
      <div className="div-titulo">
        {" "}
        <h3>{category}</h3>
        <button className="button-x" onClick={() => onDeleteCategory(category)}>
          X
        </button>
      </div>
      {isLoading ? <h2>Cargando...</h2> : null}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} /> // {...image} se usa para mandar todas las props cuando son muchas
        ))}
      </div>
    </>
  );
};
