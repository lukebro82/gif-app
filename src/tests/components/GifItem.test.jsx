import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe("Pruebas en <GifItem/>", () => {
  const title = "dragonball";
  const url = "https://dragonball.com/";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByRole("img").src).toBe(url);
    expect(screen.getByRole("img").alt).toBe(title);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
