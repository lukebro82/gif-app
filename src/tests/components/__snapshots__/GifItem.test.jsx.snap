// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="dragonball"
      src="https://dragonball.com/"
    />
    <p>
      dragonball
    </p>
  </div>
</div>
`;
